core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -235,6 +270,57 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
     }
   }
 
+
+  /**
+    * Asynchronous version of handleStateChange only for the case when targetState == OnlinePartition.
+    * For any other transitions the syncronous version should be used
+    */
+  private def handleStateChangeAsync(topic: String, partition: Int, targetState: PartitionState,
+                                     leaderSelector: PartitionLeaderSelector,
+                                     topCallback: (scala.Any) => Unit, topCtx: scala.Any) {
```

Comment:
Generally, usage of `Any` in Scala is discouraged. Can we not use a type parameter instead?